src/Utilities/ContainerHelpers.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Requires.hpp"
+
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Returns the `i`th element if `T` has a subscript operator, otherwise
+ * if `T` is fundamental returns `t`.
+ */
+template <class T>
+SPECTRE_ALWAYS_INLINE auto get_element(const T& t, const size_t i)
```

Comment:
Aren't the const versions redundant with the non const ones since `T` can match a const type?